From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: brickmonster <92665597+brickmonster@users.noreply.github.com>
Date: Sun, 14 Apr 2024 00:34:54 +0100
Subject: [PATCH] improve player chat api: inline constants


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index e77bf7f432387bdfa7f69d31b014e8cd254fd4ca..5632676e9e80d46fcad96d47f400246d13143bbf 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -665,7 +665,7 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
             if (msg.startsWith("/")) {
                 this.getHandle().connection.handleCommand(msg);
             } else {
-                final PlayerChatMessage playerChatMessage = PlayerChatMessage.system(msg).withResult(new net.minecraft.network.chat.ChatDecorator.ModernResult(Component.literal(msg), true, false));
+                final PlayerChatMessage playerChatMessage = PlayerChatMessage.system(msg).withResult(new net.minecraft.network.chat.ChatDecorator.ModernResult(Component.literal(msg)));
                 // TODO chat decorating
                 // TODO text filtering
                 this.getHandle().connection.chat(msg, playerChatMessage, false);
