From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: brickmonster <92665597+brickmonster@users.noreply.github.com>
Date: Sun, 14 Apr 2024 01:28:44 +0100
Subject: [PATCH] adventure: add Component overload of PlayerChatMessage


diff --git a/src/main/java/net/minecraft/network/chat/PlayerChatMessage.java b/src/main/java/net/minecraft/network/chat/PlayerChatMessage.java
index 81e904e78c9fec49ac244f0a62cef75e5824f2dd..714cc1a2307ee732d65dcc1a0674ba6d5de17fb8 100644
--- a/src/main/java/net/minecraft/network/chat/PlayerChatMessage.java
+++ b/src/main/java/net/minecraft/network/chat/PlayerChatMessage.java
@@ -25,6 +25,10 @@ public record PlayerChatMessage(
         final PlayerChatMessage msg = this.withUnsignedContent(result.component());
         return new PlayerChatMessage(msg.link, msg.signature, msg.signedBody, msg.unsignedContent, msg.filterMask, result);
     }
+    public PlayerChatMessage withResult(Component result) {
+        final PlayerChatMessage msg = this.withUnsignedContent(result);
+        return new PlayerChatMessage(msg.link, msg.signature, msg.signedBody, msg.unsignedContent, msg.filterMask, new net.minecraft.network.chat.ChatDecorator.ModernResult(result));
+    }
     public net.minecraft.network.chat.ChatDecorator.Result requireResult() {
         return Objects.requireNonNull(this.result, "Requires a decoration result to be set here");
     }
